File: scratch/check_db.py
import sys
sys.path.append('scripts/prospectus_scraper')
from supabase import create_client
from config import SUPABASE_URL, SUPABASE_KEY

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
row = supabase.table("ipos").select("*").eq("ticker", "JELI").execute().data[0]
ipo_id = row['id']

print("--- IPO INSIDER RISK ---")
res1 = supabase.table("ipo_insider_risk").select("*").eq("ipo_id", ipo_id).execute()
print(res1.data)

print("--- SHAREHOLDERS ---")
res2 = supabase.table("ipo_shareholders").select("*").eq("ipo_id", ipo_id).execute()
print(res2.data)

print("--- FINANCIAL HIGHLIGHTS ---")
res3 = supabase.table("ipo_financial_highlights").select("*").eq("ipo_id", ipo_id).execute()
print(res3.data)

File: scripts/check_db.js
require('dotenv').config({ path: '.env.local' });
const { createClient } = require('@supabase/supabase-js');

const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY);

async function test() {
  const { data: ipos } = await supabase.from('ipos').select('id, ticker').in('ticker', ['PRDL', 'JELI']);
  console.log("IPOs:", ipos);
  
  if (ipos && ipos.length > 0) {
    const ids = ipos.map(i => i.id);
    const { data: risks, error } = await supabase.from('ipo_insider_risk').select('*').in('ipo_id', ids);
    console.log("Risks Table Sample:");
    console.log(JSON.stringify(risks, null, 2));
  }
}
test();
